File: IClientShortcuts.cs

using System;
using System.Runtime.InteropServices;

namespace SteamStructs {
    [StructLayout(LayoutKind.Sequential, Pack = 1)]
    internal struct IClientShortcuts
    {
        public IntPtr GetUniqueLocalAppId;
        public IntPtr GetGameIDForAppID;
        public IntPtr GetAppIDForGameID;
        public IntPtr GetShortcutCount;
        public IntPtr GetShortcutAppIDByIndex;
        public IntPtr GetShortcutAppNameByIndex;
        public IntPtr GetShortcutExeByIndex;
        public IntPtr GetShortcutUserTagCountByIndex;
        public IntPtr GetShortcutUserTagByIndex;
        public IntPtr BIsShortcutRemoteByIndex;
        public IntPtr BIsTemporaryShortcutByIndex;
        public IntPtr BIsOpenVRShortcutByIndex;
        public IntPtr BIsDevkitShortcutByIndex;
        public IntPtr GetDevkitGameIDByIndex;
        public IntPtr GetDevkitAppIDByDevkitGameID;
        public IntPtr GetShortcutAppNameByAppID;
        public IntPtr GetShortcutExeByAppID;
        public IntPtr GetShortcutStartDirByAppID;
        public IntPtr GetShortcutIconByAppID;
        public IntPtr GetShortcutPathByAppID;
        public IntPtr GetShortcutCommandLine;
        public IntPtr GetShortcutUserTagCountByAppID;
        public IntPtr GetShortcutUserTagByAppID;
        public IntPtr BIsShortcutRemoteByAppID;
        public IntPtr BIsShortcutHiddenByAppID;
        public IntPtr BIsTemporaryShortcutByAppID;
        public IntPtr BIsOpenVRShortcutByAppID;
        public IntPtr BAllowDesktopConfigByAppID;
        public IntPtr BAllowOverlayByAppID;
        public IntPtr GetShortcutLastPlayedTime;
        public IntPtr AddShortcut;
        public IntPtr AddTemporaryShortcut;
        public IntPtr AddOpenVRShortcut;
        public IntPtr SetShortcutFromFullpath;
        public IntPtr SetShortcutAppName;
        public IntPtr SetShortcutExe;
        public IntPtr SetShortcutStartDir;
        public IntPtr SetShortcutIcon;
        public IntPtr SetShortcutCommandLine;
        public IntPtr ClearShortcutUserTags;
        public IntPtr AddShortcutUserTag;
        public IntPtr RemoveShortcutUserTag;
        public IntPtr ClearAndSetShortcutUserTags;
        public IntPtr SetShortcutHidden;
        public IntPtr SetAllowDesktopConfig;
        public IntPtr SetAllowOverlay;
        public IntPtr SetOpenVRShortcut;
        public IntPtr SetDevkitShortcut;
        public IntPtr RemoveShortcut;
        public IntPtr RemoveAllTemporaryShortcuts;
        public IntPtr LaunchShortcut;
    }
}

File: IClientShortcuts.cc
class IClientShortcuts {
	virtual unknown_ret GetUniqueLocalAppId() = 0;
	virtual unknown_ret GetGameIDForAppID(unsigned int) = 0;
	virtual unknown_ret GetAppIDForGameID(CGameID) = 0;
	virtual unknown_ret GetShortcutCount() = 0;
	virtual unknown_ret GetShortcutAppIDByIndex(unsigned int) = 0;
	virtual unknown_ret GetShortcutAppNameByIndex(unsigned int) = 0;
	virtual unknown_ret GetShortcutExeByIndex(unsigned int) = 0;
	virtual unknown_ret GetShortcutUserTagCountByIndex(unsigned int) = 0;
	virtual unknown_ret GetShortcutUserTagByIndex(unsigned int, unsigned int) = 0;
	virtual unknown_ret BIsShortcutRemoteByIndex(unsigned int) = 0;
	virtual unknown_ret BIsTemporaryShortcutByIndex(unsigned int) = 0;
	virtual unknown_ret BIsOpenVRShortcutByIndex(unsigned int) = 0;
	virtual unknown_ret BIsDevkitShortcutByIndex(unsigned int) = 0;
	virtual unknown_ret GetDevkitGameIDByIndex(unsigned int) = 0;
	virtual unknown_ret GetDevkitAppIDByDevkitGameID(char const*) = 0;
	virtual unknown_ret GetShortcutAppNameByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutExeByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutStartDirByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutIconByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutPathByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutCommandLine(unsigned int) = 0;
	virtual unknown_ret GetShortcutUserTagCountByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutUserTagByAppID(unsigned int, unsigned int) = 0;
	virtual unknown_ret BIsShortcutRemoteByAppID(unsigned int) = 0;
	virtual unknown_ret BIsShortcutHiddenByAppID(unsigned int) = 0;
	virtual unknown_ret BIsTemporaryShortcutByAppID(unsigned int) = 0;
	virtual unknown_ret BIsOpenVRShortcutByAppID(unsigned int) = 0;
	virtual unknown_ret BAllowDesktopConfigByAppID(unsigned int) = 0;
	virtual unknown_ret BAllowOverlayByAppID(unsigned int) = 0;
	virtual unknown_ret GetShortcutLastPlayedTime(unsigned int) = 0;
	virtual unknown_ret AddShortcut(char const*, char const*, char const*, char const*, char const*) = 0;
	virtual unknown_ret AddTemporaryShortcut(char const*, char const*, char const*) = 0;
	virtual unknown_ret AddOpenVRShortcut(char const*, char const*, char const*) = 0;
	virtual unknown_ret SetShortcutFromFullpath(unsigned int, char const*) = 0;
	virtual unknown_ret SetShortcutAppName(unsigned int, char const*) = 0;
	virtual unknown_ret SetShortcutExe(unsigned int, char const*) = 0;
	virtual unknown_ret SetShortcutStartDir(unsigned int, char const*) = 0;
	virtual unknown_ret SetShortcutIcon(unsigned int, char const*) = 0;
	virtual unknown_ret SetShortcutCommandLine(unsigned int, char const*) = 0;
	virtual unknown_ret ClearShortcutUserTags(unsigned int) = 0;
	virtual unknown_ret AddShortcutUserTag(unsigned int, char const*) = 0;
	virtual unknown_ret RemoveShortcutUserTag(unsigned int, char const*) = 0;
	virtual unknown_ret ClearAndSetShortcutUserTags(unsigned int, SteamParamStringArray_t const*) = 0;
	virtual unknown_ret SetShortcutHidden(unsigned int, bool) = 0;
	virtual unknown_ret SetAllowDesktopConfig(unsigned int, bool) = 0;
	virtual unknown_ret SetAllowOverlay(unsigned int, bool) = 0;
	virtual unknown_ret SetOpenVRShortcut(unsigned int, bool) = 0;
	virtual unknown_ret SetDevkitShortcut(unsigned int, char const*) = 0;
	virtual unknown_ret RemoveShortcut(unsigned int) = 0;
	virtual unknown_ret RemoveAllTemporaryShortcuts() = 0;
	virtual unknown_ret LaunchShortcut(unsigned int, unsigned int) = 0;
};
